### ISSUE TITLE
"Another profiler is running" in TF 2.2.0rc0

### ISSUE THREAD

**Issue by drasmuss** [tensorflow/tensorflow#37543]
**System information** 
- Have I written custom code (as opposed to using a stock
example script provided in TensorFlow): yes
- OS Platform and Distribution (e.g.,
Linux Ubuntu 16.04): Windows 10
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if
the issue happens on mobile device: N/A
- TensorFlow installed from (source or
binary): binary
- TensorFlow version (use command below): 2.2.0rc0
- Python version: 3.6.10
- Bazel
version (if compiling from source): N/A
- GCC/Compiler version (if compiling from
source): N/A
- CUDA/cuDNN version: N/A
- GPU model and memory: N/A

**Describe the current behavior**

"Another profiler is running" error when using the TensorBoard callback profiling twice with `profile_batch=1` in non-eager mode.

**Describe the expected behavior**

There should be no error.

**Standalone code to reproduce the issue** 
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.
``` python
import tensorflow as tf
import numpy as np

tf.compat.v1.disable_eager_execution()

inp = tf.keras.Input((1,))
out = tf.keras.layers.Dense(units=1)(inp)
model = tf.keras.Model(inp, out)

model.compile(optimizer=tf.optimizers.SGD(1), loss=tf.losses.mse)

model.fit(
    np.zeros((64, 1)),
    np.zeros((64, 1)),
    callbacks=[tf.keras.callbacks.TensorBoard(log_dir="tmp", profile_batch=1)],
)

model.fit(
    np.zeros((64, 1)),
    np.zeros((64, 1)),
    callbacks=[tf.keras.callbacks.TensorBoard(log_dir="tmp", profile_batch=1)],
)
```

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.
```
Traceback (most recent call last):
  File ".../tmp.py", line 21, in <module>
    callbacks=[tf.keras.callbacks.TensorBoard(log_dir="tmp", profile_batch=1)],
  File "...\lib\site-packages\tensorflow\python\keras\callbacks.py", line 1736, in __init__
    profiler.warmup()  # Improve the profiling accuracy.
  File "...\lib\site-packages\tensorflow\python\profiler\profiler_v2.py", line 124, in warmup
    start('')
  File "...\lib\site-packages\tensorflow\python\profiler\profiler_v2.py", line 74, in start
    'Another profiler is running.')
tensorflow.python.framework.errors_impl.AlreadyExistsError: Another profiler is running.
```

**Comment by gadagashwini** [comment]
I could replicate the issue with TF 2.2.rc0.
Please find the gist [here](https://colab.sandbox.google.com/gist/gadagashwini/2c9946a562d22127690069c4013da8bc/untitled449.ipynb). Thanks!

**Comment by ckluk** [comment]
Looks like the issue here is that the profiling in the first model.fit() is still active while the profiling in the second model.fit() tries to start profiling. This shouldn't happen if the first model.fit() is done. Let me look into it in more details.

**Comment by drasmuss** [comment]
The problem is this block here https://github.com/tensorflow/tensorflow/blob/v2.2.0-rc0/tensorflow/python/keras/callbacks.py#L1950, which starts the profiler regardless of whether eager mode is enabled or not, but then the profiler is only stopped if eager mode is enabled https://github.com/tensorflow/tensorflow/blob/v2.2.0-rc0/tensorflow/python/keras/callbacks.py#L1980.

**Comment by qiuminxu** [comment]
Verifies the fix in tf-nightly works in this colab.
https://colab.research.google.com/gist/gadagashwini/2c9946a562d22127690069c4013da8bc/untitled449.ipynb

Cherrypick request sent out:
https://github.com/tensorflow/tensorflow/pull/37599

**Comment by goldiegadde** [comment]
#37599  has been merged into the r2.2 branch, this fix should be available in the next rc candidate.

**Comment by qiuminxu** [comment]
@drasmuss Can you try again and see if the error is fixed? Thanks!

**Comment by drasmuss** [comment]
Yes this is fixed in TF2.2rc2

**Comment by omya3** [comment]
Still facing the issue

**Comment by jdserranomu** [comment]
Still facing the issue.

**Comment by yisitu** [comment]
@omya3, @jdserranomu could you clarify the version of TensorFlow where you are seeing the issue?

**Comment by jdserranomu** [comment]
The version I am running is 2.4.0.

**Comment by GF-Huang** [comment]
Issue still 2.4.1.

**Comment by ryol8888** [comment]
Still facing the issue in tf-nightly

**Comment by jbaiocchi** [comment]
Can you try with the new nightly? The following commit makes the callback resilient to concurrent profiling:
https://github.com/tensorflow/tensorflow/commit/53ff2339b3c05a6efbe5bdbdd6c40b8973fb8e17

**Comment by yisitu** [comment]
Friendly ping @drasmuss

**Comment by drasmuss** [comment]
What are you looking for from me?

**Comment by yisitu** [comment]
My bad, wrong ping. @omya3

**Comment by apdullahyayik** [comment]
Just update the callbacks for the second fit, issue goes away

**Comment by sachinprasadhs** [comment]
The issue seems to be resolved in Tf Nightly 2.6.0-dev20210525, please find the gist [here](https://colab.research.google.com/gist/sachinprasadhs/ed6bdbd914579557e7c3380656081887/untitled7.ipynb).

**Comment by nomurakeiya** [comment]
when I used Tensorflow-GPU2.4.1, faced same error.
Then I changed to 2.6.0, faced another error.
*training code starts, but NaN occurs during training.
https://github.com/matterport/Mask_RCNN/issues/2670
![Untitled](https://user-images.githubusercontent.com/56009331/130339683-2d8b6433-8d3c-495b-957a-8c338c42ec0d.png)


Now I using TensorFlow-GPU2.5.0, there aren't error!!!
following below, my environment.


windows 11 Pro (dev)
Python 3.7.7
NVIDIA GeForce RTX3090
TensorFlow-GPU 2.5.0
cuDNN 8.2.2 forCUDA11.4
CUDA Toolkit 11.4.0
